backend/gsr_booking/management/commands/send_gsr_reminders.py

```diff
@@ -0,0 +1,46 @@
+import datetime
+
+from django.core.management.base import BaseCommand
+from django.utils import timezone
+
+from gsr_booking.models import Reservation
+from user.models import SERVICE_GSR_BOOKING, NotificationToken
+from user.notifications import send_push_notif
+
+
+class Command(BaseCommand):
+    help = "Sends reminders for the GSR Bookings."
+
+    def handle(self, *args, **kwargs):
+        # iterate through all reservations scheduled for the next 30 minutes
+        for reservation in Reservation.objects.filter(
+            start__gt=timezone.now(), start__lte=timezone.now() + datetime.timedelta(minutes=30)
+        ):
+            if reservation.is_cancelled:
+                continue
+
+            creator = reservation.creator
+
+            # NOTE: only iOS for now, change when Android gets set up
+            token = (
+                NotificationToken.objects.filter(user=creator, kind=NotificationToken.KIND_IOS)
+                .exclude(token="")
+                .first()
+            )
+
+            if token:
+                # skip user if their setting = disabled for GSR Notifs
+                setting = token.objects.filter(service=SERVICE_GSR_BOOKING).first()
+                if setting and not setting.enabled:
```

Comment:
for this if statement, i'm not sure if it also takes care of if the setting is non-existent. maybe the conditional could be like `if not setting or not settings.enabled`